packages/rum-recorder/src/domain/segmentCollection.ts

```diff
@@ -0,0 +1,138 @@
+import { CreationReason, MouseMoveRecord, Record, RecordType, SegmentContext, SegmentMeta } from '../types'
+import { getRecordStartEnd, groupMouseMoves, isMouseMoveRecord } from './recordUtils'
+
+export const MAX_SEGMENT_DURATION = 30_000
+export const MAX_MOUSE_MOVE_BATCH = 100
+
+export interface SegmentWriter {
+  write(data: string): void
+  complete(data: string, meta: SegmentMeta): void
+}
+
+export function startSegmentCollection(getSegmentContext: () => SegmentContext | undefined, writer: SegmentWriter) {
+  let currentSegment: Segment | undefined
+
+  renewSegment('init')
+
+  function renewSegment(creationReason: CreationReason) {
+    if (currentSegment) {
+      currentSegment.finish()
```

Comment:
* 'flush' is only used in a test description, I'll change it to 'write' since it is more accurate
* ok to rename `segment.finish` to `segment.complete`
* I'm not sure about `renew` though: it indicates that the a new segment is created.

Reply:
> 'flush' is only used in a test description

it is also used in comment in `trackSegmentRenewal` when using renewSegment.
FMU, flush is when we empty the buffer of the segment and to send it to the server. It corresponds more to complete usages, no?

Reply:
FMU, flush, complete, renew seems to be different intentions but it produces the same result.
If we want to create a new segment, it flushes the current one.
If we want to flush the current segment, it creates a new one.
It seems really close to the lifecycle of the batch concept used in other packages, it could be nice to be a bit more similar if it makes sense.

Reply:
FTR: let's rename 'renew' and 'complete' to 'flush'